src/orion/core/io/database/mongodb.py

```diff
@@ -90,6 +89,9 @@ def __init__(self, host='localhost', name=None,
 
         super(MongoDB, self).__init__(host, name, port, username, password)
```

Comment:
The authentification errors are because of `initiate_connection()` in the Parent class. You can solve this by passing `authSource` and `serverSelectionTimeoutMS` directly to `__init__` to get them set in `options`. By the way, I would use `serverSelectionTimeoutMS` in the interface instead of `server_timeout` to keep it coherent with `pymongo` and because it is specific to the MongoDB backend.